uni/lib/view/common_widgets/toast_message.dart

```diff
@@ -1,17 +1,144 @@
 import 'package:flutter/material.dart';
-import 'package:toast/toast.dart';
+import 'package:flutter/cupertino.dart';
 
 /// Provides feedback about an operation in a small popup
 ///
 /// usage example: ToastMessage.display(context, toastMsg);
-class ToastMessage {
-  static const Color toastColor = Color.fromARGB(255, 100, 100, 100);
-  static display(BuildContext context, String msg) {
-    ToastContext().init(context);
-    Toast.show(msg,
-        duration: Toast.lengthLong,
-        gravity: Toast.bottom,
-        backgroundColor: toastColor,
-        backgroundRadius: 16.0);
+class MToast extends StatelessWidget {
+  final String message;
+  final Color? color;
+  final IconData? icon;
+  final Color? iconColor;
+  final Color? textColor;
+  final AlignmentGeometry? alignment;
+  final dynamic elevation;
+
+  const MToast(
+      {Key? key,
+        required this.message,
+        this.color = Colors.white,
+        this.icon,
+        this.iconColor = Colors.black,
+        this.textColor = Colors.black,
+        this.alignment = Alignment.bottomCenter,
+        this.elevation = 0.0})
+      : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+      onWillPop: () async => false,
+      child: Dialog(
+        insetPadding: const EdgeInsets.all(50),
+        alignment: alignment,
+        backgroundColor: color,
+        shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(10.0))),
+        elevation: elevation,
+        child: Row(
+          mainAxisSize: MainAxisSize.min,
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Container(
+                margin: const EdgeInsets.all(10.0),
+                child: icon != null ? Icon(
+                  icon,
+                  color: iconColor,
+                ) : const SizedBox.shrink()),
```

Comment:
You can make the icon a required parameter and delete this conditional since you are not adding methods without icons